crates/ruff_python_ast/src/types.rs

```diff
@@ -76,3 +79,61 @@ impl<'a> From<&RefEquality<'a, Expr>> for &'a Expr {
 }
 
 pub type CallPath<'a> = smallvec::SmallVec<[&'a str; 8]>;
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct Import {
+    pub name: String,
+    pub location: Location,
+    pub end_location: Location,
+}
+
+#[derive(Debug, Clone, Default, Serialize, Deserialize)]
+pub struct Imports {
+    inner: FxHashMap<PathBuf, Vec<Import>>,
+}
+
+impl Imports {
+    pub fn insert(&mut self, module: PathBuf, imports_vec: Vec<Import>) {
+        self.inner.insert(module, imports_vec);
+    }
+
+    pub fn extend(&mut self, other: Self) {
+        self.inner.extend(other.inner);
+    }
+
+    pub fn expand_relative(
+        modules: &[&str],
+        module: &Option<String>,
+        name: &str,
+        level: &Option<usize>,
+    ) -> String {
+        match level {
+            Some(level) => {
+                // last part of path is always the module itself
+                let resolved_module = if *level > 0 {
+                    format!(
+                        "{}.",
+                        modules
+                            .iter()
+                            .rev()
+                            .take(*level)
+                            .collect::<Vec<_>>()
+                            .iter()
+                            .rev()
+                            .map(|s| (**s).to_string())
+                            .collect::<Vec<String>>()
+                            .join(".")
+                    )
+                } else {
+                    String::new()
+                };
+                if let Some(module_name) = module.as_ref() {
+                    format!("{resolved_module}{module_name}.{name}")
+                } else {
+                    format!("{resolved_module}{name}")
+                }
+            }
+            None => format!("{}.{}", module.as_ref().unwrap_or(&String::new()), name),
```

Comment:
Can we look at, or reuse, or share logic with `crates/ruff/src/rules/flake8_tidy_imports/relative_imports.rs#fix_banned_relative_import`? I believe it's solving the same problem.